tools/publish_release.py

```diff
@@ -1,76 +1,49 @@
 #!/usr/bin/env python
-# for more info see https://developer.github.com/v3/repos/releases/
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import json
 import os
 from lib.github import GitHub
+from lib.helpers import *
 import requests
 
-BROWSER_LAPTOP_REPO = 'brave/browser-laptop'
-TARGET_ARCH= os.environ['TARGET_ARCH'] if os.environ.has_key('TARGET_ARCH') else 'x64'
 RELEASE_NAME = 'Dev Channel Beta'
 
 def main():
-  github = GitHub(auth_token())
-  releases = github.repos(BROWSER_LAPTOP_REPO).releases.get()
-  version = json.load(open('package.json'))['version']
-  tag = ('v' + version + release_channel())
-  tag_exists = False
-  for release in releases:
-    if not release['draft'] and release['tag_name'] == tag:
-      tag_exists = True
-      break
-  release = create_or_get_release_draft(github, releases, tag,
-                                        tag_exists)
-  # match version to GitHub milestone
-  commit_tag = None
-  parts = version.split('.', 3)
-  if (len(parts) == 3):
-    parts[2] = 'x'
-    commit_tag = '.'.join(parts)
-
-  # Press the publish button.
-  if not tag_exists and commit_tag:
-    publish_release(github, release['id'], tag, commit_tag)
-
-def create_release_draft(github, tag):
-  name = '{0} {1}'.format(RELEASE_NAME, tag)
-  # TODO: Parse release notes from CHANGELOG.md
-  body = '(placeholder)'
-  if body == '':
-    sys.stderr.write('Quit due to empty release note.\n')
-    sys.exit(1)
-  data = dict(tag_name=tag, name=name, body=body, draft=True, prerelease=True)
-  r = github.repos(BROWSER_LAPTOP_REPO).releases.post(data=data)
-  return r
+  repo = GitHub(get_env('GITHUB_TOKEN')).repos(BROWSER_LAPTOP_REPO)
 
-def create_or_get_release_draft(github, releases, tag, tag_exists):
-  # Search for existing draft.
-  for release in releases:
-    if release['draft']:
-      return release
+  release = get_draft(repo, get_tag())
+  commit_tag = get_commit_tag(get_version())
 
-  if tag_exists:
-    tag = 'do-not-publish-me'
-  return create_release_draft(github, tag)
+  print("Found draft {}, release this? [y/n]".format(release['tag_name']))
+  if raw_input() != 'y':
```

Comment:
I haven't run this myself, but this assumes the user is manually executing this script. Is that the only way its done today?

Reply:
I thought just the upload was automated, not publishing?

Reply:
Didn't see anything running it in jenkins at least, is it being run somewhere else?

My thinking is we do automate the publish part it will at least be gated.

Reply:
The whole process is automated actually- in jenkins.brave.com, you can view the following jobs:
- browser-laptop-build-linux
- browser-laptop-build-macos
- browser-laptop-build-win32
- browser-laptop-build-win64

Each of them runs the `./tools/cibuild.py` script which does npm install, builds binary, packages binary, and then uploads to GitHub

Reply:
The existing process works fine as-is, except the fact that tags aren't considered when searching for the build on GitHub. Instead of blindly using any draft, it should also use the tag name

Reply:
Sorry for the delay, got that removed.